### PR TITLE
SDCICD-87. Addon test harness support.

### PR DESCRIPTION
**Pull request by meowfaceman**
Addon test harnesses can now be supplied to osde2e. This will allow for
addon authors to test against OSD clusters. Results from addons will be
collected by osde2e for later analysis.

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -130,6 +138,9 @@ type TestConfig struct {
 	// OperatorSkip is a comma-delimited list of operator names to ignore health checks from. ex. "insights,telemetry"
 	OperatorSkip string `env:"OPERATOR_SKIP" sect:"tests" default:"insights" yaml:"ginkgoFocus"`
 
+	// SkipClusterHealthChecks skips the cluster health checks. Useful when developing against a running cluster.
+	SkipClusterHealthChecks bool `env:"SKIP_CLUSTER_HEALTH_CHECKS" sect:"tests" yaml:"skipClusterHealthChecks"`
```
Comment (jeefy) [inline pkg/config/config.go]: I think it's better to specify `default:"false"` here

File: pkg/runner/command.go [hunk 2]
```diff
@@ -7,7 +7,7 @@ import (
 )
 
 // testCmd configures default Service Account as a kubeconfig, runs openshift-tests, and serves results over HTTP
-const testCmd = `
+const testCmd = `#!/bin/bash
```
Comment (jeefy) [inline pkg/runner/command.go]: We should explicitly call this out and ensure documentation states the test container images require `bash` to be installed. 

Often times containers don't actually have `bash` installed, only sh. This isn't so much a critique on the PR but raising the need to document this once addon testing starts.

File: pkg/config/config.go (hunk 1 above)
Comment (meowfaceman) [reply pkg/config/config.go]: Good catch -- had meant to do this.

File: pkg/runner/command.go (hunk 2 above)
Comment (meowfaceman) [reply pkg/runner/command.go]: Agreed here -- I'll add this to the doc.

**Comment by meowfaceman** [comment]
/hold cancel
Saw some must-gather failures, but those aren't new. This looks good.

**Pull request merged**